magellan-library/src/main/java/com/wealthfront/magellan/transitions/CircularRevealTransition.kt

```diff
@@ -7,17 +7,23 @@ import android.os.Build
 import android.view.View
 import android.view.ViewAnimationUtils
 import android.view.ViewGroup
+import androidx.annotation.IdRes
 import androidx.interpolator.view.animation.FastOutSlowInInterpolator
 import com.wealthfront.magellan.Direction
 import kotlin.math.hypot
 
+@Suppress("FunctionName")
+public fun CircularRevealTransition(clickedView: View): MagellanTransition {
+  return CircularRevealTransition(clickedView.id)
+}
+
 /**
  * A [MagellanTransition] that reveals the next screen circularly outward from the middle of the
- * [clickedView].
+ * [clickedViewId].
  *
- * @property clickedView the view on which this circular reveal is centered
+ * @property clickedViewId the id of the view on which this circular reveal is centered
  */
-public class CircularRevealTransition(private val clickedView: View) : MagellanTransition {
+private class CircularRevealTransition(@IdRes private val clickedViewId: Int) : MagellanTransition {
```

Comment:
Hold the id instead of the view to avoid leak

Reply:
nice. Sure, I guess we can assume the ID will be unique